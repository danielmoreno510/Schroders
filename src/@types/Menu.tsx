/* eslint-disable */
type Page = string;
