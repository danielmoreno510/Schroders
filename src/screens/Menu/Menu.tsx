import React from 'react';
import {
  SafeAreaView,
  ScrollView,
  TouchableOpacity,
  View,
  Text,
} from 'react-native';

import styles from './styles';
import {INSET, menuOptions} from '../../constants';
import {NativeBaseProvider} from 'native-base';

const Menu: React.FC<Navigation> = ({navigation}) => {
  const changeScreeen = (screen: Page) => () => navigation.navigate(screen);

  return (
    <SafeAreaView style={styles.safeAreaContainer}>
      <ScrollView>
        <NativeBaseProvider initialWindowMetrics={INSET}>
          {menuOptions.map(option => (
            <View key={option.name} style={styles.optionContainer}>
              <TouchableOpacity
                testID={option.name}
                style={styles.option}
                onPress={changeScreeen(option.name)}>
                <View style={styles.rowContainer}>
                  <option.icon />
                  <Text style={styles.textOption}>{option.name}</Text>
                </View>
              </TouchableOpacity>
            </View>
          ))}
        </NativeBaseProvider>
      </ScrollView>
    </SafeAreaView>
  );
};

export default Menu;
