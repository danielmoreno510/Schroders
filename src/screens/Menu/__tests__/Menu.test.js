import React from 'react';
import TestRenderer from 'react-test-renderer';

import Menu from '../Menu';

describe('Menu component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      navigation: {navigate: jest.fn()},
    };

    describe('WHEN the component is rendered', () => {
      const testRenderer = TestRenderer.create(<Menu {...testProps} />);

      it('THEN should display a regular Menu', () => {
        expect(testRenderer.toJSON()).toMatchSnapshot();
      });
    });

    describe('WHEN the menu option is pressed', () => {
      const testRenderer = TestRenderer.create(<Menu {...testProps} />);
      const testInstance = testRenderer.root;
      testInstance.findByProps({testID: 'Home'}).props.onPress();

      it('THEN navigation function should be called', () => {
        expect(testProps.navigation.navigate).toBeCalledTimes(1);
      });
    });
  });
});
