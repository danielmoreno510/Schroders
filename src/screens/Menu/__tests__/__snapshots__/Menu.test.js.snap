// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Menu 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <RNCSafeAreaProvider
        onInsetsChange={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "paddingHorizontal": 16,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
                "paddingVertical": 12,
              }
            }
            testID="Home"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <RNSVGSvgView
                accessibilityRole="image"
                align="xMidYMid"
                bbHeight="16"
                bbWidth="16"
                dataSet={{}}
                focusable={false}
                meetOrSlice={0}
                minX={0}
                minY={0}
                stroke=""
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "color": "#737373",
                      "height": 16,
                      "width": 16,
                    },
                    {
                      "flex": 0,
                      "height": 16,
                      "width": 16,
                    },
                  ]
                }
                tintColor="#737373"
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                    ]
                  }
                  stroke={null}
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGPath
                      d="M12 0C5.376 0 0 5.376 0 12C0 18.624 5.376 24 12 24C18.624 24 24 18.624 24 12C24 5.376 18.624 0 12 0ZM9.6 17.4V6.6L16.8 12L9.6 17.4Z"
                      fill={
                        {
                          "type": 2,
                        }
                      }
                      propList={
                        [
                          "fill",
                          "stroke",
                        ]
                      }
                      stroke={null}
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  {
                    "color": "black",
                    "fontSize": 16,
                    "paddingHorizontal": 10,
                  }
                }
              >
                Home
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "paddingHorizontal": 16,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
                "paddingVertical": 12,
              }
            }
            testID="Commit List"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <RNSVGSvgView
                accessibilityRole="image"
                align="xMidYMid"
                bbHeight="16"
                bbWidth="16"
                dataSet={{}}
                focusable={false}
                meetOrSlice={0}
                minX={0}
                minY={0}
                stroke=""
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "color": "#737373",
                      "height": 16,
                      "width": 16,
                    },
                    {
                      "flex": 0,
                      "height": 16,
                      "width": 16,
                    },
                  ]
                }
                tintColor="#737373"
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                    ]
                  }
                  stroke={null}
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGPath
                      d="M18.0621 16.1487C17.2993 16.1487 16.6113 16.4451 16.0895 16.9223L8.93445 12.7495C8.98988 12.523 9.02482 12.2868 9.02482 12.0458C9.02482 11.8048 8.98988 11.5698 8.93445 11.3433L16.0136 7.21147C16.5679 7.73368 17.3006 8.02461 18.0621 8.02482C19.7237 8.02482 21.0745 6.67406 21.0745 5.01241C21.0745 3.35076 19.7237 2 18.0621 2C16.4004 2 15.0496 3.35076 15.0496 5.01241C15.0496 5.2534 15.0846 5.48958 15.14 5.71491L8.06085 9.84673C7.50628 9.32504 6.7738 9.0342 6.01241 9.03338C4.35077 9.03338 3 10.3841 3 12.0458C3 13.7074 4.35077 15.0582 6.01241 15.0582C6.77392 15.058 7.50661 14.7671 8.06085 14.2449L15.2159 18.4176C15.1632 18.633 15.1361 18.8538 15.1352 19.0756C15.1354 19.6541 15.3072 20.2196 15.6288 20.7005C15.9503 21.1814 16.4073 21.5562 16.9418 21.7775C17.4764 21.9988 18.0645 22.0567 18.6319 21.9437C19.1993 21.8308 19.7205 21.5522 20.1296 21.1431C20.5387 20.734 20.8173 20.2128 20.9302 19.6454C21.0432 19.078 20.9853 18.4898 20.764 17.9553C20.5427 17.4208 20.1679 16.9638 19.687 16.6423C19.2061 16.3207 18.6406 16.1489 18.0621 16.1487Z"
                      fill={
                        {
                          "type": 2,
                        }
                      }
                      propList={
                        [
                          "fill",
                          "stroke",
                        ]
                      }
                      stroke={null}
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  {
                    "color": "black",
                    "fontSize": 16,
                    "paddingHorizontal": 10,
                  }
                }
              >
                Commit List
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "paddingHorizontal": 16,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
                "paddingVertical": 12,
              }
            }
            testID="About"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <RNSVGSvgView
                accessibilityRole="image"
                align="xMidYMid"
                bbHeight="16"
                bbWidth="16"
                dataSet={{}}
                focusable={false}
                meetOrSlice={0}
                minX={0}
                minY={0}
                stroke=""
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "color": "#737373",
                      "height": 16,
                      "width": 16,
                    },
                    {
                      "flex": 0,
                      "height": 16,
                      "width": 16,
                    },
                  ]
                }
                tintColor="#737373"
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                    ]
                  }
                  stroke={null}
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGGroup
                      fill={
                        {
                          "type": 2,
                        }
                      }
                      propList={
                        [
                          "fill",
                          "stroke",
                        ]
                      }
                      stroke={null}
                    >
                      <RNSVGPath
                        d="M11.9442 8.24C12.5628 8.24 13.0642 7.73856 13.0642 7.12C13.0642 6.50144 12.5628 6 11.9442 6C11.3257 6 10.8242 6.50144 10.8242 7.12C10.8242 7.73856 11.3257 8.24 11.9442 8.24Z"
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                      />
                      <RNSVGPath
                        d="M14.4008 15.9996H12.8008V9.59961H10.4008C10.1886 9.59961 9.98513 9.68389 9.8351 9.83392C9.68507 9.98395 9.60078 10.1874 9.60078 10.3996C9.60078 10.6118 9.68507 10.8153 9.8351 10.9653C9.98513 11.1153 10.1886 11.1996 10.4008 11.1996H11.2008V15.9996H9.60078C9.38861 15.9996 9.18513 16.0839 9.0351 16.2339C8.88507 16.384 8.80078 16.5874 8.80078 16.7996C8.80078 17.0118 8.88507 17.2153 9.0351 17.3653C9.18513 17.5153 9.38861 17.5996 9.60078 17.5996H14.4008C14.613 17.5996 14.8164 17.5153 14.9665 17.3653C15.1165 17.2153 15.2008 17.0118 15.2008 16.7996C15.2008 16.5874 15.1165 16.384 14.9665 16.2339C14.8164 16.0839 14.613 15.9996 14.4008 15.9996Z"
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                      />
                      <RNSVGPath
                        d="M12 2C10.0222 2 8.08879 2.58649 6.4443 3.6853C4.79981 4.78412 3.51809 6.3459 2.76121 8.17316C2.00433 10.0004 1.8063 12.0111 2.19215 13.9509C2.578 15.8907 3.53041 17.6725 4.92894 19.0711C6.32746 20.4696 8.10929 21.422 10.0491 21.8078C11.9889 22.1937 13.9996 21.9957 15.8268 21.2388C17.6541 20.4819 19.2159 19.2002 20.3147 17.5557C21.4135 15.9112 22 13.9778 22 12C22 9.34783 20.9464 6.80429 19.0711 4.92893C17.1957 3.05357 14.6522 2 12 2ZM12 20.3333C10.3518 20.3333 8.74066 19.8446 7.37025 18.9289C5.99984 18.0132 4.93174 16.7117 4.30101 15.189C3.67028 13.6663 3.50525 11.9908 3.82679 10.3742C4.14834 8.75774 4.94201 7.27288 6.10745 6.10744C7.27288 4.942 8.75774 4.14833 10.3743 3.82679C11.9908 3.50525 13.6663 3.67027 15.189 4.301C16.7117 4.93173 18.0132 5.99984 18.9289 7.37025C19.8446 8.74066 20.3333 10.3518 20.3333 12C20.3333 13.0943 20.1178 14.178 19.699 15.189C19.2802 16.2001 18.6664 17.1187 17.8926 17.8926C17.1187 18.6664 16.2001 19.2802 15.189 19.699C14.178 20.1178 13.0944 20.3333 12 20.3333Z"
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  {
                    "color": "black",
                    "fontSize": 16,
                    "paddingHorizontal": 10,
                  }
                }
              >
                About
              </Text>
            </View>
          </View>
        </View>
      </RNCSafeAreaProvider>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
